core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -31,14 +31,17 @@
 
 class BaseFileScanTask implements FileScanTask {
   private final DataFile file;
+  private final DeleteFile[] deletes;
   private final String schemaString;
   private final String specString;
   private final ResidualEvaluator residuals;
 
   private transient PartitionSpec spec = null;
 
-  BaseFileScanTask(DataFile file, String schemaString, String specString, ResidualEvaluator residuals) {
+  BaseFileScanTask(DataFile file, DeleteFile[] deletes, String schemaString, String specString,
```

Comment:
This is mostly for my understanding: is `DeleteFile[] deletes`  a mandatory builder param now for file scan tasks? If not, from a v1 / v2 compatibility standpoint would it make sense to add an overloaded constructor?

Reply:
This implementation and constructor are internal and package-private. The public API is the `FileScanTask` interface. I went ahead and updated all of the places where we use this, so we don't need to have an overloaded constructor. I think it is better to do it this way so that we ensure that the parameter is always explicit and we aren't accidentally ignoring delete files anywhere.